<a href="https://colab.research.google.com/github/jfjoung/AI_For_Chemistry/blob/main/notebooks/Week2_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Week 2-1: Python 기초 (변수, 리스트, for문)


## 파이썬 코드는 "위에서 아래로" 실행됩니다

- 셀을 실행하면, 그 셀의 코드가 **위에서 아래 순서로** 실행됩니다.
- 아직 정의되지 않은 변수를 사용하면 에러가 납니다.
- 에러는 "틀렸다"가 아니라, **컴퓨터가 이해하지 못했다**는 뜻입니다.

👉 오늘 목표:
- Google Colab에서 파이썬 코드를 실행할 수 있다
- 변수를 사용해 숫자 계산을 할 수 있다
- 리스트(list)를 이용해 화학 데이터를 저장할 수 있다
- for문을 사용해 간단한 화학 계산을 수행할 수 있다
- "에러를 무서워하지 않고, 직접 고쳐본다"

중요
- 코드는 반드시 `Shift + Enter`로 실행하세요
- 에러가 나면 당황하지 말고, 마지막 줄 에러 메시지를 먼저 읽으세요
- 오늘 가장 흔한 실수: `:` 빠뜨리기, 들여쓰기 오류

In [ ]:
# 실행 확인용 코드
print("Week 2-1 수업을 시작합니다.")


Week 2-1 수업을 시작합니다.


In [ ]:
# 이 코드는 에러가 납니다. 왜일까요?
print(x)


In [ ]:
x = 10
print(x)


## 1. 변수(variable)와 숫자 계산

변수란?
- 어떤 값을 저장하는 상자
- `=` 는 같다는 의미가 아니라 "저장한다"는 뜻

파이썬에서는 계산기를 쓰듯이 계산할 수 있습니다.

## 변수는 "물리량 / 화학량"의 이름표가 될 수 있습니다.

변수는 단순한 숫자가 아니라,
- 온도
- 압력
- 농도
- 분자량
같은 **화학적 의미를 가진 값**을 저장합니다.


In [ ]:
# 변수 예제
temperature_K = 298.15
pressure_atm = 1.0
concentration_M = 0.5

print("온도 (K):", temperature_K)
print("압력 (atm):", pressure_atm)
print("농도 (M):", concentration_M)

온도 (K): 298.15
압력 (atm): 1.0
농도 (M): 0.5


In [ ]:
# 간단한 계산
print("온도 + 10 =", temperature_K + 10)
print("압력 × 2 =", pressure_atm * 2)

온도 + 10 = 308.15
압력 × 2 = 2.0


In [ ]:
# 단위 변환 예
pressure_bar = pressure_atm * 1.01325
print("압력 (bar):", pressure_bar)

압력 (bar): 1.01325


### 학생 실습 1

섭씨 25도를 Kelvin으로 변환해봅시다.

공식:
- Kelvin = Celsius + 273.15

아래 코드에서 빈칸(`___`)을 채우고 실행하세요.


In [ ]:
# TODO: 빈칸을 채우세요
temp_C = ___
temp_K = temp_C + ___

print("Kelvin 온도:", temp_K)


### 학생 실습 2
아래 코드의 문제점을 찾아보세요.

```
temperature = 25
print("온도 (K):", temperature)
```




질문:

- 이 온도는 K일까? °C일까?
- 코드상에서 구분이 되는가?
- temperature_K 같은 변수 이름이
  그냥 a, b 보다 좋은 이유는 무엇일까요?


### 이상기체 상태방정식으로 압력 계산하기

In [ ]:
R = 0.082057  # L·atm / (mol·K)
n = 1.0       # mol
T = 298.15    # K
V = 24.5      # L

P = n * R * T / V
print("계산된 압력 (atm):", P)


계산된 압력 (atm): 0.9985834510204081


## 리스트는 "순서가 있는 데이터"

중요한 특징:
- 순서가 있다
- 0번부터 시작한다

예:
- 분자 내 원자 순서
- 실험 데이터 (시간 순서)
- 여러 샘플의 측정값

In [ ]:
#인덱스
elements = ["H", "C", "O", "N"]

print(elements)
print("0번:", elements[0])
print("1번:", elements[1])
print("마지막:", elements[-1])


['H', 'C', 'O', 'N']
0번: H
1번: C
마지막: N


In [ ]:
# 리스트 예제
elements = ["H", "C", "O", "N"]
atomic_weights = [1.008, 12.01, 16.00, 14.01]

print("원소 리스트:", elements)
print("원자량 리스트:", atomic_weights)

print("첫 번째 원소:", elements[0])
print("첫 번째 원자량:", atomic_weights[0])
print("산소(O)의 원자량:", atomic_weights[2])


원소 리스트: ['H', 'C', 'O', 'N']
원자량 리스트: [1.008, 12.01, 16.0, 14.01]
첫 번째 원소: H
첫 번째 원자량: 1.008
산소(O)의 원자량: 16.0


### 학생 실습 3

이산화탄소 CO<sub>2</sub>는 다음과 같이 구성됩니다.
- C 1개
- O 2개

CO<sub>2</sub> 를 원소 문자열 리스트로 만들어 출력해보세요.


In [ ]:
# TODO: CO2 원소 리스트 만들기
co2_atoms = ___
print(co2_atoms)


### 학생 실습 4 (인덱스 연습)

다음 리스트에서:
- 첫 번째 원소
- 마지막 원소

를 각각 출력해보세요.


In [ ]:
atoms = ["Na", "Cl", "H", "O"]

# TODO
print(atoms[___])
print(atoms[___])


이 코드는 에러가 날까?

결과는 “과학적으로” 맞을까?

아래 코드의 결과는 맞을까요?  

틀리다면, 왜 틀렸는지 생각해보세요.


In [ ]:
elements = ["H", "C", "O", "N"]
atomic_weights = [12.01, 1.008, 16.00, 14.01]

print("첫 번째 원소:", elements[0])
print("첫 번째 원소의 원자량:", atomic_weights[0])


### 학생 실습 5

아래 리스트에서 항상 마지막 원소를 출력해보세요.


In [ ]:
molecules = ["H2O", "CO2", "NH3", "CH4"]

# TODO
print("마지막 분자:", __________)


생각해봅시다.

아래 정보는 리스트로 표현하는 게 좋을까요?

- 온도: 298 K
- 압력: 1 atm
- 농도: 0.5 M

왜 그렇게 생각하나요?


## 3. for문 – 반복 작업하기

for문은 리스트 안의 값을 하나씩 꺼내서 반복 작업을 할 때 사용합니다.

기본 형태:
```python
for 항목 in 리스트:
    실행할 코드
```

주의:

반드시 : 를 써야 합니다

다음 줄은 반드시 들여쓰기 해야 합니다.


## 왜 for문이 필요한가?

리스트 길이가 늘어나면?
- 직접 하나씩 쓰는 것은 불가능
- "반복"이 필요함

for문 = "리스트 안의 모든 값에 대해 같은 작업 수행"


In [ ]:
# for문 없이 (비효율적)
values = [1, 2, 3, 4]

print(values[0])
print(values[1])
print(values[2])
print(values[3])


In [ ]:
# for문 사용 (권장)
for v in values:
    print(v)


In [ ]:
# for문 예제
for element in elements:
    print("원소:", element)

원소: H
원소: C
원소: O
원소: N


In [ ]:
temperatures = [25, 30, 40]
for T in temperatures:
    print(T, "°C")


25 °C
30 °C
40 °C


### 학생 실습 6

아래 리스트의 값만 바꿔도  
출력 코드는 그대로 유지되는지 확인해보세요.


In [ ]:
values = [10, 20, 30, 40, 50]

for v in values:
    print(v)


10
20
30
40
50


### 학생 실습 7

아래 온도 조건에서  
각 온도를 K 단위로 변환해 출력해보세요.


In [ ]:
temperatures_C = [25, 30, 40]

for T in temperatures_C:
    print(T + 273.15)


298.15
303.15
313.15


### 학생 실습 8

출력 결과가 무엇을 의미하는지  
사람이 읽을 수 있게 출력해보세요.


In [ ]:
temperatures_C = [25, 30, 40]

for T in temperatures_C:
    print("온도:", T, "°C")


온도: 25 °C
온도: 30 °C
온도: 40 °C


### 학생 실습 9

아래 코드를 실행해보고,  
에러 메시지를 읽고 고쳐보세요.


In [ ]:
values = [1, 2, 3]

for v in values
print(v)


### 학생 실습 10

아래 분자의 원소들을 하나씩 출력해보세요.


In [ ]:
co2_atoms = ["C", "O", "O"]
for ____ in co2_atoms:
    print("원소: ", __)

다음 작업에서 for문이 필요한지 생각해보세요.

1. 온도 하나 출력
2. 온도 3개 출력
3. 온도 100개 출력

어디서부터 for문이 필요할까요?


## 4. 합계와 평균 계산하기

for문을 사용하면 여러 값의 합계를 계산할 수 있습니다.


In [ ]:
# 원자량 합계 계산
total = 0
for w in atomic_weights:
    total = total + w

print("원자량 합계:", total)

average = total / len(atomic_weights)
print("원자량 평균:", average)


원자량 합계: 43.028
원자량 평균: 10.757


### 학생 실습 11

다음 리스트의 합계를 for문으로 계산해보세요.

리스트:
- values = [2, 4, 6, 8, 10]

조건:
- for문을 반드시 사용할 것


In [ ]:
# TODO: for문을 사용해 합계 계산
values = [2, 4, 6, 8, 10]

total = ___
for v in ___:
    ___

print("합계:", total)


## 오늘의 핵심 정리

- 리스트: 분자를 구성하는 원소들을 저장
- for문: 분자 안의 원소들을 하나씩 처리
- 변수 + 반복문 = 분자량 계산의 기초

다음 시간에는:
- 조건문(if)
- 함수(function)
를 이용해 분류해볼 예정입니다.
